torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -417,6 +417,7 @@ def __init__(
                 "forward_prefetch": forward_prefetch,
                 "limit_all_gathers": limit_all_gathers,
                 "use_orig_params": use_orig_params,
+                "ignored_states": self._ignored_params,
```

Comment:
why do we need to propagate ignored_states, but not ignored_modules?

Reply:
Good point. Maybe we should also propagate `ignored_modules`. This might be another bug.

Reply:
The reason not propagating `ignored_modules` does not break correctness today is because we _only_ use `ignored_modules` to compute the ignored parameters and to compute which modules to ignore for auto wrapping. 

For those two functionalities, we do not need the nested FSDP instances to have `ignored_modules`. However, we should probably still propagate it in case we use `ignored_modules` for something else in the future.

Reply:
I see. In general, what do you think about having both ignored_states and ignored_modules? this seems like it can get confusing to users, shall we just consolidate to ignored_states?